### ISSUE TITLE
Gallery Widget: migrate to new Core Gallery Widget

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#7830]
WordPress Core is looking into adding a new Gallery Widget in WP 4.9:
https://core.trac.wordpress.org/ticket/41914

It would be nice if we could migrate [our own Gallery Widget](https://jetpack.com/support/extra-sidebar-widgets/gallery-widget/) over, while keeping its options.

**Comment by westonruter** [comment]
The core gallery widget has landed in trunk (4.9-alpha). See [r41590](https://core.trac.wordpress.org/changeset/41590).

The Tiled Galleries module and slideshows in Jetpack should work properly with core's Gallery widget, but this is [reported](https://wptavern.com/new-core-gallery-widget-targeted-for-wordpress-4-9) to not be working right:

> For example, with Jetpack enabled, users can choose between a thumbnail and a slideshow gallery, but the slideshow option doesn’t seem to work correctly in the widget.

Dev note: https://make.wordpress.org/core/2017/09/25/introducing-the-gallery-widget/

**Comment by westonruter** [comment]
Core patch has been proposed to improve extensibility of the media widgets: https://core.trac.wordpress.org/ticket/42285

With that patch committed, all Jetpack would have to do is:

```php
add_filter( 'widget_media_gallery_instance_schema', function( $schema ) {
	$schema['type'] = array(
		'type' => 'string',
		'enum' => array( 'default', 'rectangular', 'square', 'circle', 'columns' ),
		'description' => __( 'Type of gallery.', 'jetpack' ),
		'default' => 'default',
	);
	return $schema;
} );
```

Please test.
